src/measure_theory/vector_measure.lean

```diff
@@ -429,6 +429,169 @@ end measure
+variables {v : vector_measure α M}
+
+/-- The pushforward of a vector measure along a measurable function. -/
+def map (v : vector_measure α M) (f : α → β) :
+  vector_measure β M :=
+if hf : measurable f then
+{ measure_of' := λ s, if measurable_set s then v (f ⁻¹' s) else 0,
+  empty' := by simp,
+  not_measurable' := λ i hi, if_neg hi,
+  m_Union' :=
+  begin
+    intros g hg₁ hg₂,
+    convert v.m_Union (λ i, hf (hg₁ i)) (λ i j hij x hx, hg₂ i j hij hx),
+    { ext i, rw if_pos (hg₁ i) },
+    { rw [preimage_Union, if_pos (measurable_set.Union hg₁)] }
+  end } else 0
+
+lemma map_apply {f : α → β} (hf : measurable f) {s : set β} (hs : measurable_set s) :
+  v.map f s = v (f ⁻¹' s) :=
+by { rw [map, dif_pos hf], exact if_pos hs }
+
+@[simp] lemma map_id : v.map id = v :=
+ext (λ i hi, by rw [map_apply measurable_id hi, preimage_id])
+
+@[simp] lemma map_zero (f : α → β) : (0 : vector_measure α M).map f = 0 :=
+begin
+  by_cases hf : measurable f,
+  { ext i hi,
+    rw [map_apply hf hi, zero_apply, zero_apply] },
+  { exact dif_neg hf }
+end
+
+/-- The restriction of a vector measure on some measurable set-/
+def restrict (v : vector_measure α M) (i : set α) :
+  vector_measure α M :=
+if hi : measurable_set i then
+{ measure_of' := λ s, if measurable_set s then v (s ∩ i) else 0,
+  empty' := by simp,
+  not_measurable' := λ i hi, if_neg hi,
+  m_Union' :=
+  begin
+    intros f hf₁ hf₂,
+    convert v.m_Union (λ n, (hf₁ n).inter hi)
+      (hf₂.mono $ λ i j, disjoint.mono inf_le_left inf_le_left),
+    { ext n, rw if_pos (hf₁ n) },
+    { rw [Union_inter, if_pos (measurable_set.Union hf₁)] }
+  end } else 0
+
+lemma restrict_apply {i : set α} (hi : measurable_set i)
+  {j : set α} (hj : measurable_set j) : v.restrict i j = v (j ∩ i) :=
+by { rw [restrict, dif_pos hi], exact if_pos hj }
+
+lemma restrict_eq_self {i : set α} (hi : measurable_set i)
+  {j : set α} (hj : measurable_set j) (hij : j ⊆ i) : v.restrict i j = v j :=
+by rw [restrict_apply hi hj, inter_eq_left_iff_subset.2 hij]
+
+@[simp] lemma restrict_empty : v.restrict ∅ = 0 :=
+ext (λ i hi, by rw [restrict_apply measurable_set.empty hi, inter_empty, v.empty, zero_apply])
+
+@[simp] lemma restrict_univ : v.restrict univ = v :=
+ext (λ i hi, by rw [restrict_apply measurable_set.univ hi, inter_univ])
+
+@[simp] lemma restrict_zero {i : set α} :
+  (0 : vector_measure α M).restrict i = 0 :=
+begin
+  by_cases hi : measurable_set i,
+  { ext j hj, rw [restrict_apply hi hj], refl },
+  { exact dif_neg hi }
+end
+
+section has_continuous_add
+
+variables [has_continuous_add M]
+
+lemma map_add (v w : vector_measure α M) (f : α → β) :
+  (v + w).map f = v.map f + w.map f :=
+begin
+  by_cases hf : measurable f,
+  { ext i hi,
+    simp [map_apply hf hi] },
+  { simp [map, dif_neg hf] }
+end
+
+/-- `vector_measure.map` as an additive monoid homomorphism. -/
+@[simps] def map' (f : α → β) : vector_measure α M →+ vector_measure β M :=
```

Comment:
Instead of using a prime, could you maybe name it `map_gm` (for "group morphism")? Same thing for `restrict'` below. And the linear version could be `map_lm` or `mapₗ`, I think we have both conventions.

Reply:
`measure.restrict` uses the second convention with the subscript (https://leanprover-community.github.io/mathlib_docs/measure_theory/measure_space.html#measure_theory.measure.restrict%E2%82%97) so I chose the second convention for linear maps